source/overview/faq.rst

```diff
@@ -187,6 +187,31 @@ Does Mattermost have an official website-based plug-in to offer anonymous chat t
 
+
+Design Decisions
+----------------
+
+Why does Mattermost give an error message when they incorrectly enter a password? 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Mattermost's core design principle is to be `"fast, obvious, forgiving" <https://docs.mattermost.com/developer/fx-guidelines.html#fast-obvious-forgiving>`_ and tellings users that they made a mistake in entering their password is in service of our principle of prioritizing user interests. 
+
+When using username-password authentication, especially with AD/LDAP, there's the possibility of usernames being email addresses, Mattermost username, AD/LDAP username, AD/LDAP ID or other AD/LDAP attributes and our design principle intends to help end users understand if their login error came from having the wrong password or the wrong email/username.
+
+We believe this approach reduces geniune user login attempt errors, making it more obvious when malicious login attempt errors occur when monitoring system logs. 
+
+We also believe this reduces the number of password reset requests, which reduces opportunities for attackers to spoof or intercept requests. 
+
+Finally, we believe this reduces the number of help desk tickets an organization encounters, which reclaims productivity and lowers costs. 
+
+Moreover, the approach to notifying a user when their password is incorrectly entered is a growing in popularity, with business applications like G Suite that serve tens of millions of users in this way. 
+
+The trade-off with this design is that if physical security is not in effect, and network security is not effect (i.e. no VPN, or a situation where an attacker has penetrated the private network), and username-password authentication is used, it is easier for an attacker to enumerate email addresses or usernames by sending HTTP requests to the system, up to the maximum number of requests per second defined in Mattermost's `API rate limiting settings <https://docs.mattermost.com/administration/config-settings.html#rate-limiting>`_. 
+
+For organizations who choose to deploy in such a configuration, please consider the following mitigations: 1) instead of username-password, use a single-sign-on provider in Mattermost Enterprise Edition like OneLogin, Okta or ADFS, use the open source GitLab SSO option available with Mattermost Team Edition, 2) per the recommended install instructions, use a VPN client to apply network security to your deployment, 3) enable monitoring and alerting from your proxy server to detect and isolate malicious behavior reaching your deployment. 
+
+Above all, make sure to subscribe to the `Mattermost Security Bulletin <https://about.mattermost.com/security-bulletin/>`_ and apply security patches as recommended.  
+
```

Comment:
I'm just gonna go ahead and say I'm 5/5 on not including any arguments that make it sound like we're doing things this way for the sake of better security. They're highly debatable at best, and to those with different stances, may reflect poorly on the company and the individuals responsible for our security (e.g. me).

I would suggest keeping the first two paragraphs about principles / UX, saying "The possible security implications and reduced privacy are outweighed by these values and are considered acceptable risk.", and closing it with the usual security recommendations. But please don't say things like "if network security is not in effect, this makes it easier for attackers to exploit this...". This represents increased attack surface regardless of whether the deployment is behind a firewall or not. They're layered defenses, not alternative defenses.